### PR TITLE
3.4.6 - 13505 - Deck band select fix

### PR DESCRIPTION
**Pull request by Cattlesquat**
There has been a thread on the forums (http://www.vassalengine.org/forum/viewtopic.php?f=3&t=11438) where they are complaining about it doing a band-select when you drag things off a deck. 

For some reason this problem seems "worse" on Macs than on PC, but I am able to see at least some of the problem on PC. 

In any event, I this this PR should fix the problem. Doesn't allow band-selects to start on top of a deck.

### CODE REVIEW

**Comment by BrentEaston** [comment]
You seem to have made this fix against Master not Release 3-4-6?

**Comment by Cattlesquat** [comment]
Oh thanks for pointing that out -- I actually made it against Release 3.4.6 but hadn't marked that in the PR.

**Comment by BrentEaston** [comment]
Where you able to test this on a Mac Brian? Otherwise, we will need to get a snapshot build for the Mac guys to try out.

**Comment by Cattlesquat** [comment]
I didn't end up borrowing my wife's Mac, because I was actually seeing the band-select problem happen in Windows using that guy's module (no idea why his does it a lot worse than PoG, but anyway). So I'm certainly not against a snapshot build though I'm not sure it's actually necessary -- since problem was happening for me and now it's not.

File: vassal-app/src/main/java/VASSAL/build/module/map/KeyBufferer.java [hunk 1]
```diff
@@ -104,16 +104,22 @@ public void mousePressed(MouseEvent e) {
 
     final KeyBuffer kbuf = KeyBuffer.getBuffer();
 
-    GamePiece p = map.findPiece(e.getPoint(), PieceFinder.PIECE_IN_STACK);
+    GamePiece p = map.findPiece(e.getPoint(), PieceFinder.DECK_OR_PIECE_IN_STACK);
     // Don't clear the buffer until we find the clicked-on piece
     // Because selecting a piece affects its visibility
     EventFilter filter = null;
     BandSelectType bandSelect = BandSelectType.NONE;
+    Boolean isDeck = (p != null) && (p instanceof Deck);
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/map/KeyBufferer.java:112]: Just make this a `boolean`. No need for the object version of it.

**Review by uckelman — CHANGES_REQUESTED** [review]

**Pull request merged**